src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -269,10 +347,8 @@ func (s *peersSource) fetchBootstrapBlocksFromPeers(
 			s.logFetchBootstrapBlocksFromPeersOutcome(shard, shardResult, err)
 
 			if err != nil {
-				// Do not add result at all to the bootstrap result
-				lock.Lock()
-				bootstrapResult.Add(shard, nil, xtime.NewRanges(currRange))
-				lock.Unlock()
+				// No result to add for this bootstrap.
+				unfulfill(currRange)
```

Comment:
I may be misunderstanding the flow here, but wouldn't this be unfulfilled already? Or is this to revert anything fulfilled for partial failure cases?

Reply:
Yeah unless you explicitly set unfulfilled it won't be fulfilled.